From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kreijstal <elektrischrainbow@gmail.com>
Date: Mon, 26 Oct 2026 04:27:03 +0100
Subject: [PATCH 3/4] Keep using the bundled qm-dsp with --use-external-libs

--use-external-libs switches libltc, fluidsynth, hidapi, the VAMP SDK
and qm-dsp to system copies. Distributions want the first four, but
qm-dsp is not packaged for MSYS2 and upstream qm-dsp >= 1.8 is not API
compatible with Ardour's VAMP plugins anyway (Arch Linux carries the
same change as ardour-7.0-re-vendor_qm-dsp.patch).

Always configure and build libs/qm-dsp, and stop adding the unused
--qm-dsp-include directory (default /usr/include/qm-dsp) to CXXFLAGS.
---
 libs/qm-dsp/wscript | 6 +++---
 wscript             | 2 --
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/libs/qm-dsp/wscript b/libs/qm-dsp/wscript
index 01b4601..66965b6 100644
--- a/libs/qm-dsp/wscript
+++ b/libs/qm-dsp/wscript
@@ -14,6 +14,9 @@ def options(opt):
     pass
 
 def configure(conf):
+    # Always use the bundled copy: Ardour's VAMP plugins are written
+    # against qm-dsp 1.7.x and system qm-dsp >= 1.8 is not compatible.
+    return
     if conf.is_defined('USE_EXTERNAL_LIBS'):
         conf.check_cxx(header_name="base/Pitch.h", mandatory=True, msg="Checking for qm-dsp headers")
         conf.check_cxx(lib="qm-dsp", uselib_store="QMDSP", mandatory=True)
@@ -48,9 +51,6 @@ def configure(conf):
                            )
 
 def build(bld):
-    if bld.is_defined('USE_EXTERNAL_LIBS'):
-        return
-
     # Host Library
     obj = bld(features = 'c cxx cxxstlib')
     obj.source = '''
diff --git a/wscript b/wscript
index e43e9f2..ecf3393 100644
--- a/wscript
+++ b/wscript
@@ -1134,8 +1134,6 @@ def configure(conf):
 
     if Options.options.use_external_libs:
         conf.define('USE_EXTERNAL_LIBS', 1)
-        conf.env.append_value(
-            'CXXFLAGS', '-I' + Options.options.qm_dsp_include)
 
     if Options.options.boost_include != '':
         conf.env.append_value('CXXFLAGS', '-I' + Options.options.boost_include)
